packages/core/test/financial-templates/optimistic-distributor/OptimisticDistributor.js

```diff
@@ -0,0 +1,908 @@
+            rewardToken.options.address,
+            customAncillaryData
+          )
+          .send({ from: sponsor }),
+        "OO liveness too small"
+      )
+    );
+
+    // optimisticOracleLivenessTime exactly at MINIMUM_LIVENESS should be accepted.
+    await optimisticDistributor.methods
+      .createReward(
+        rewardAmount,
+        earliestProposalTimestamp,
+        bondAmount,
+        minimumLiveness,
+        identifier,
+        rewardToken.options.address,
+        customAncillaryData
+      )
+      .send({ from: sponsor });
+
+    // Fund sponsor for creating new rewards.
+    await mintAndApprove(rewardToken, sponsor, optimisticDistributor.options.address, rewardAmount, deployer);
+
+    // optimisticOracleLivenessTime exactly at MAXIMUM_LIVENESS should revert.
+    assert(
+      await didContractRevertWith(
+        optimisticDistributor.methods
+          .createReward(
+            rewardAmount,
+            earliestProposalTimestamp,
+            bondAmount,
+            maximumLiveness,
+            identifier,
+            rewardToken.options.address,
+            customAncillaryData
+          )
+          .send({ from: sponsor }),
+        "OO liveness too large"
+      )
+    );
+
+    // optimisticOracleLivenessTime just below MAXIMUM_LIVENESS should be accepted.
+    await optimisticDistributor.methods
+      .createReward(
+        rewardAmount,
+        earliestProposalTimestamp,
+        bondAmount,
+        maximumLiveness - 1,
+        identifier,
+        rewardToken.options.address,
+        customAncillaryData
+      )
+      .send({ from: sponsor });
+
+    // Fund sponsor for creating new rewards.
+    await mintAndApprove(rewardToken, sponsor, optimisticDistributor.options.address, rewardAmount, deployer);
+
+    // Fetch balances before creating new reward.
+    const sponsorBalanceBefore = toBN(await rewardToken.methods.balanceOf(sponsor).call());
+    const contractBalanceBefore = toBN(
+      await rewardToken.methods.balanceOf(optimisticDistributor.options.address).call()
+    );
+
+    // Fetch expected next rewardIndex.
+    const rewardIndex = parseInt(await optimisticDistributor.methods.nextCreatedReward().call());
+
+    // Create new reward.
+    const receipt = await optimisticDistributor.methods
+      .createReward(...defaultRewardParameters)
+      .send({ from: sponsor });
+
+    // Fetch balances after creating new reward.
+    const sponsorBalanceAfter = toBN(await rewardToken.methods.balanceOf(sponsor).call());
+    const contractBalanceAfter = toBN(
+      await rewardToken.methods.balanceOf(optimisticDistributor.options.address).call()
+    );
+
+    // Check for correct change in balances.
+    assert.equal(sponsorBalanceBefore.sub(sponsorBalanceAfter).toString(), rewardAmount);
+    assert.equal(contractBalanceAfter.sub(contractBalanceBefore).toString(), rewardAmount);
+
+    // Check that created rewards are emitted.
+    await assertEventEmitted(
```

Comment:
very good. thank you for being detailed focused in your testing! I know some of these can be a pain to write.